_playwright-tests/UI/PopularRepo.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { closePopupsIfExist } from './helpers/helpers';
+import { deleteAllPopularRepos } from './helpers/deletePopularRepositories';
+
+test.describe('Popular Repositories', () => {
+  test('Add popular repos', async ({ page }) => {
+    // Ensure no popular repos are selected.
+    await deleteAllPopularRepos(page);
+
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+
+    await expect(page).toHaveTitle('Repositories - Content | RHEL');
+    await page.getByRole('link', { name: 'Popular repositories' }).click();
+    await expect(page.getByTestId('popular_repos_table')).toBeVisible();
+
+    // Select the first row then use main side toggle menu to add it
+    await page
+      .getByRole('row', { name: 'EPEL 9 Everything x86_64' })
+      .getByLabel('Select row 0', { exact: true })
+      .click();
+    // Open the drop-down menu
+    await page.getByTestId('add-selected-dropdown-toggle-no-snap').click();
+    // Click the first and only button in the drop-down menu
+    await page.getByTestId('add-selected-repos-no-snap').click();
+    // Multiple toast pop-ups about snapshotting and the code tries to close that
+    await page
+      .getByRole('row', { name: 'EPEL 8 Everything x86_64' })
+      .getByLabel('Select row 1', { exact: true })
+      .click();
+    // Open the drop-down menu
+    await page.getByTestId('add-selected-dropdown-toggle-no-snap').click();
+    // Click the first and only button in the drop-down menu
+    await page.getByTestId('add-selected-repos-no-snap').click();
+    // Multiple toast pop-ups about snapshotting and the code tries to close that
+    // Move to Custom repo tab
+    await page.getByRole('link', { name: 'Your repositories' }).click();
+    await expect(page.getByTestId('custom_repositories_table')).toBeVisible();
+
+    // Filter for EPEL and pick the first row
```

Comment:
All of these could be `steps` and having it commented in that way

Reply:
Done that now